mpt/impl/imap-mailbox/core/src/main/java/org/apache/james/mpt/imapmailbox/suite/ACLCommands.java

```diff
@@ -19,68 +19,87 @@
 
 package org.apache.james.mpt.imapmailbox.suite;
 
+import java.util.Locale;
+
 import org.apache.james.mailbox.model.MailboxACL;
 import org.apache.james.mailbox.model.MailboxPath;
 import org.apache.james.mailbox.model.SimpleMailboxACL;
-import org.apache.james.mpt.api.HostSystem;
+import org.apache.james.mpt.api.ImapHostSystem;
 import org.apache.james.mpt.imapmailbox.GrantRightsOnHost;
-import org.apache.james.mpt.imapmailbox.suite.base.BaseImapProtocol;
+import org.apache.james.mpt.imapmailbox.ImapTestConstants;
+import org.apache.james.mpt.imapmailbox.MailboxMessageAppender;
+import org.junit.Before;
 import org.junit.Test;
 
-import javax.inject.Inject;
-import java.util.Locale;
-
-public class ACLCommands extends BaseImapProtocol {
+public abstract class ACLCommands implements ImapTestConstants {
+    
     public static final String OTHER_USER_NAME = "Boby";
     public static final String OTHER_USER_PASSWORD = "password";
     public static final MailboxPath OTHER_USER_MAILBOX = new MailboxPath("#private", OTHER_USER_NAME, "") ;
-    private final MailboxACL.MailboxACLRights readWriteSeenRight;
 
-    @Inject
-    private static HostSystem system;
-    @Inject
+    protected abstract ImapHostSystem createImapHostSystem();
+    protected abstract GrantRightsOnHost createGrantRightsOnHost();
+    
+    private ImapHostSystem system;
     private GrantRightsOnHost grantRightsOnHost;
 
-    public ACLCommands() throws Exception {
-        super(system);
+    private MailboxACL.MailboxACLRights readWriteSeenRight;
+    private ACLScriptedTestProtocol scriptedTestProtocol;
```

Comment:
It needs some methods to interact with the HostSystem that most servers don't have. Actually, only cyrus has that.
Putting these methods into IMAPScriptedTestProtocol and having only partial implementation is IMO not as good because you discover that methods are not implemented at runtime vs compile time with my solution.